pipenv/core.py

```diff
@@ -1115,10 +1115,20 @@ def do_lock(
             normalized = pep423_name(dep['name'])
             if not hasattr(dep, 'keys') or not hasattr(dep['name'], 'keys'):
                 continue
-            is_top_level = dep['name'] in vcs_lockfile or normalized in vcs_lockfile
-            lockfile_key = next(k for k in [dep['name'], normalized] if k in vcs_lockfile)
-            lockfile_entry = vcs_lockfile[lockfile_key] if is_top_level else None
-            dep_lockfile = clean_resolved_dep(dep, is_top_level=is_top_level, pipfile_entry=lockfile_entry)
+            is_top_level = (
+                dep['name'] in vcs_lockfile or
```

Comment:
I should mention this naming is wrong now, `vcs_lockfile` comes from code that doesn't do any resolving anymore.  This is the only place in our codebase that passes deps to the resolver. This is technically `vcs_pipfile` and should be renamed -- that's what makes something a top-level entry... presence in the pipfile.... That's partly the value of this pr